Generate serializes for ObjectIdentifier (by adding the ability to generate serializers for template specializations)

beidson opened the pull request:
#### ef78eaa134b5c063abbe9664a9b365eca1ef82e8
<pre>
Generate serializes for ObjectIdentifier (by adding the ability to generate serializers for template specializations)
<a href="https://bugs.webkit.org/show_bug.cgi?id=262251">https://bugs.webkit.org/show_bug.cgi?id=262251</a>
rdar://116159092

Reviewed by Alex Christensen.

This seems like a huge, complicated patch just to add generated serialization of what is essentially a uint64_t wrapper.

And it is.

But in doing so, it let me flesh out adding support for templates to generate-serializers.py
Lack of template support is something we&apos;ve had to work around before, but no more! Hopefully!

(Sadly this also had to touch all the sites for identifiers that tagged with a raw enum,
as raw enums cannot be forward declared)

* Source/JavaScriptCore/runtime/Microtask.h:
(): Deleted.
* Source/WTF/wtf/ObjectIdentifier.h:
(WTF::ObjectIdentifierGenericBase::encode const): Deleted.
(WTF::ObjectIdentifierGenericBase::decode): Deleted.
(WTF::ObjectIdentifierGeneric::decode): Deleted.
* Source/WebCore/Modules/cache/DOMCacheIdentifier.h:
(): Deleted.
* Source/WebCore/Modules/fetch/FetchIdentifier.h:
(): Deleted.
* Source/WebCore/Modules/filesystemaccess/FileSystemHandleIdentifier.h:
(): Deleted.
* Source/WebCore/Modules/filesystemaccess/FileSystemSyncAccessHandleIdentifier.h:
(): Deleted.
* Source/WebCore/Modules/filesystemaccess/WorkerFileSystemStorageConnectionCallbackIdentifier.h:
(): Deleted.
* Source/WebCore/Modules/mediastream/RTCRtpScriptTransformer.h:
* Source/WebCore/Modules/permissions/MainThreadPermissionObserverIdentifier.h:
(): Deleted.
* Source/WebCore/Modules/push-api/PushSubscriptionIdentifier.h:
(): Deleted.
* Source/WebCore/Modules/speech/SpeechRecognitionConnectionClientIdentifier.h:
(): Deleted.
* Source/WebCore/Modules/web-locks/WebLockIdentifier.h:
(): Deleted.
* Source/WebCore/Modules/websockets/WebSocketIdentifier.h:
(): Deleted.
* Source/WebCore/accessibility/AXCoreObject.h:
* Source/WebCore/dom/BroadcastChannelIdentifier.h:
(): Deleted.
* Source/WebCore/dom/ElementIdentifier.h:
(): Deleted.
* Source/WebCore/dom/PortIdentifier.h:
(): Deleted.
* Source/WebCore/editing/TextManipulationItemIdentifier.h:
(): Deleted.
* Source/WebCore/editing/TextManipulationToken.h:
* Source/WebCore/editing/cocoa/AttributedString.h:
* Source/WebCore/page/GlobalWindowIdentifier.h:
(): Deleted.
* Source/WebCore/page/SecurityOriginData.h:
(): Deleted.
* Source/WebCore/page/cocoa/ImageOverlayDataDetectionResultIdentifier.h:
(): Deleted.
* Source/WebCore/platform/MediaSessionGroupIdentifier.h:
(): Deleted.
* Source/WebCore/platform/MediaUniqueIdentifier.h:
(): Deleted.
* Source/WebCore/platform/ProcessIdentifier.h:
(): Deleted.
* Source/WebCore/platform/graphics/ImageDecoderIdentifier.h:
(): Deleted.
* Source/WebCore/platform/graphics/InbandGenericCueIdentifier.h:
(): Deleted.
* Source/WebCore/platform/graphics/LayerHostingContextIdentifier.h:
(): Deleted.
* Source/WebCore/platform/graphics/RenderingResourceIdentifier.h:
(): Deleted.
* Source/WebCore/platform/graphics/mac/DisplayConfigurationMonitor.h:
* Source/WebCore/platform/mediastream/RTCDataChannelLocalIdentifier.h:
(): Deleted.
* Source/WebCore/platform/mediastream/RealtimeMediaSourceIdentifier.h:
(): Deleted.
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCSocketIdentifier.h:
(): Deleted.
* Source/WebCore/platform/text/TextCheckingRequestIdentifier.h:
(): Deleted.
* Source/WebCore/workers/service/ServiceWorkerIdentifier.h:
(): Deleted.
* Source/WebCore/workers/service/ServiceWorkerTypes.h:
(): Deleted.
* Source/WebCore/workers/service/background-fetch/BackgroundFetchRecordIdentifier.h:
(): Deleted.
* Source/WebCore/workers/shared/SharedWorkerIdentifier.h:
(): Deleted.
* Source/WebCore/worklets/WorkletGlobalScope.h:
* Source/WebKit/GPUProcess/media/RemoteMediaResourceIdentifier.h:
(): Deleted.
* Source/WebKit/GPUProcess/media/RemoteSourceBufferIdentifier.h:
(): Deleted.
* Source/WebKit/GPUProcess/media/TrackPrivateRemoteIdentifier.h:
(): Deleted.
* Source/WebKit/NetworkProcess/CustomProtocols/LegacyCustomProtocolID.h:
(): Deleted.
* Source/WebKit/NetworkProcess/Downloads/DownloadID.h:
(): Deleted.
* Source/WebKit/NetworkProcess/PrivateClickMeasurement/cocoa/PrivateClickMeasurementNetworkLoaderCocoa.mm:
(): Deleted.
* Source/WebKit/Platform/IPC/Connection.h:
* Source/WebKit/Platform/IPC/DaemonCoders.cpp:
(WebKit::Daemon::Coder&lt;WebCore::PushSubscriptionIdentifier&gt;::encode):
(WebKit::Daemon::Coder&lt;WebCore::PushSubscriptionIdentifier&gt;::decode):
* Source/WebKit/Scripts/generate-serializers.py:
(Template):
(Template.__init__):
(Template.forward_declaration):
(Template.specialization):
(SerializedType.__init__):
(SerializedType.should_skip_forward_declare):
(one_argument_coder_declaration):
(argument_coder_declarations):
(generate_header):
(construct_type):
(construct_type.is):
(generate_one_impl):
(generate_impl):
(parse_serialized_types):
* Source/WebKit/Scripts/webkit/tests/GeneratedSerializers.cpp:
(IPC::ArgumentCoder&lt;WebKit::TemplateTest&lt;WebKit::Fabulous&gt;&gt;::encode):
(IPC::ArgumentCoder&lt;WebKit::TemplateTest&lt;WebKit::Fabulous&gt;&gt;::decode):
(IPC::ArgumentCoder&lt;WebKit::TemplateTest&lt;WebCore::Amazing&gt;&gt;::encode):
(IPC::ArgumentCoder&lt;WebKit::TemplateTest&lt;WebCore::Amazing&gt;&gt;::decode):
(IPC::ArgumentCoder&lt;WebKit::TemplateTest&lt;JSC::Incredible&gt;&gt;::encode):
(IPC::ArgumentCoder&lt;WebKit::TemplateTest&lt;JSC::Incredible&gt;&gt;::decode):
(IPC::ArgumentCoder&lt;WebKit::TemplateTest&lt;Testing::StorageSize&gt;&gt;::encode):
(IPC::ArgumentCoder&lt;WebKit::TemplateTest&lt;Testing::StorageSize&gt;&gt;::decode):
* Source/WebKit/Scripts/webkit/tests/GeneratedSerializers.h:
* Source/WebKit/Scripts/webkit/tests/SerializedTypeInfo.cpp:
(WebKit::allSerializedTypes):
* Source/WebKit/Scripts/webkit/tests/TestSerializedType.serialization.in:
* Source/WebKit/Shared/ContentWorldShared.h:
(): Deleted.
* Source/WebKit/Shared/Extensions/WebExtensionControllerIdentifier.h:
(): Deleted.
* Source/WebKit/Shared/ShapeDetectionIdentifier.h:
(): Deleted.
* Source/WebKit/Shared/WTFArgumentCoders.serialization.in:
* Source/WebKit/Shared/WebGPU/WebGPUIdentifier.h:
(): Deleted.
* Source/WebKit/Shared/WebsiteData/QuotaIncreaseRequestIdentifier.h:
(): Deleted.
* Source/WebKit/Shared/XR/XRDeviceIdentifier.h:
(): Deleted.
* Source/WebKit/UIProcess/UserMediaPermissionRequestManagerProxy.h:
* Source/WebKit/UIProcess/WebURLSchemeHandler.cpp:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/WebProcess/GPU/graphics/GraphicsContextGLIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/graphics/RenderingBackendIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/media/RemoteAudioHardwareListenerIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/media/RemoteAudioSessionIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/media/RemoteCDMIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/media/RemoteCDMInstanceIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/media/RemoteCDMInstanceSessionIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/media/RemoteLegacyCDMIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/media/RemoteLegacyCDMSessionIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/media/RemoteMediaSourceIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/media/RemoteRemoteCommandListenerIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/media/RemoteVideoFrameIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/webrtc/AudioMediaStreamTrackRendererInternalUnitIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/webrtc/MediaRecorderIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/webrtc/SampleBufferDisplayLayerIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/webrtc/VideoDecoderIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/webrtc/VideoEncoderIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCResolverIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/Storage/RetrieveRecordResponseBodyCallbackIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/WebStorage/StorageAreaImplIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/WebStorage/StorageAreaMapIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/WebStorage/StorageNamespaceIdentifier.h:
(): Deleted.

Canonical link: <a href="https://commits.webkit.org/268623@main">https://commits.webkit.org/268623@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/84f228ff773575108fc2ca621960a1efc7d1bc09

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/20183 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/20617 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/21245 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/22082 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/18834 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/20414 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/23860 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/20780 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/20291 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/20402 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/20306 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/17542 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/22933 "Built successfully") | 
| [❌ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/20369 "webkitpy-tests (failure)") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/17475 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/18342 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/24612 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/17555 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18552 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/18518 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/22583 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/19547 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/19104 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/16221 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/23589 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/18308 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/5667 "Found 6 jsc stress test failures: stress/sampling-profiler-microtasks.js.bytecode-cache, stress/sampling-profiler-microtasks.js.dfg-eager, stress/sampling-profiler-microtasks.js.eager-jettison-no-cjit, stress/sampling-profiler-microtasks.js.mini-mode, stress/sampling-profiler-microtasks.js.no-llint ...") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4844 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/22650 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/24846 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/18929 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/5491 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/JavaScriptCore/runtime/Microtask.h:
```diff
@@ -33,7 +33,7 @@ namespace JSC {
 class JSGlobalObject;
 
-enum MicrotaskIdentifierType { };
+enum class MicrotaskIdentifierType { };
```
These could also just be a struct.  Some of them are already.

achristensen07 commented on Source/WebKit/Shared/WTFArgumentCoders.serialization.in:
```diff
@@ -52,6 +54,141 @@ header: <wtf/text/AtomString.h>
+template: struct WebCore::WebSocketFrame
+template: struct WebKit::ConnectionAsyncReplyHandler
+template: struct WebKit::StorageAreaIdentifierType
+[WebKitPlatform, CustomHeader, AdditionalEncoder=StreamConnectionEncoder] class WTF::AtomicObjectIdentifier {
```
It would be nice if we could make a way for AdditionalEncoder to only apply to the ones that need it.

achristensen07 commented on Source/WebKit/Shared/WTFArgumentCoders.serialization.in:
```diff
@@ -52,6 +54,141 @@ header: <wtf/text/AtomString.h>
+template: enum class WebKit::WCLayerTreeHostIdentifierType
+template: enum class WebKit::WebExtensionControllerIdentifierType
+template: enum class WebKit::WebGPUIdentifierType
+template: enum class WebKit::WebsiteDataType : uint32_t
```
I don't see ObjectIdentifier<WebKit::WebsiteDataType> or ObjectIdentifier<WebsiteDataType> anywhere.  Is this really needed?  (along with the special parsing logic)

beidson replied:
I only added ones that needed to be there for the build.

Will double check.

beidson replied:
Was, indeed, not needed.

Removed, but leaving the storage type logic + its test, because it's gonna be relevant soon.

achristensen07 commented on Source/WebKit/Scripts/generate-serializers.py:
```diff
@@ -49,8 +49,25 @@
 
+
+class Template(object):
+    def __init__(self, template_type, namespace, name, enum_storage=None):
```
We'll likely need condition here at some point.  Maybe also just for certain instantiations.  Can be done later.

beidson committed:
Generate serializes for ObjectIdentifier (by adding the ability to generate serializers for template specializations)
https://bugs.webkit.org/show_bug.cgi?id=262251
rdar://116159092

Reviewed by Alex Christensen.

This seems like a huge, complicated patch just to add generated serialization of what is essentially a uint64_t wrapper.

And it is.

But in doing so, it let me flesh out adding support for templates to generate-serializers.py
Lack of template support is something we've had to work around before, but no more! Hopefully!

(Sadly this also had to touch all the sites for identifiers that tagged with a raw enum,
as raw enums cannot be forward declared)

* Source/JavaScriptCore/runtime/Microtask.h:
(): Deleted.
* Source/WTF/wtf/ObjectIdentifier.h:
(WTF::ObjectIdentifierGenericBase::encode const): Deleted.
(WTF::ObjectIdentifierGenericBase::decode): Deleted.
(WTF::ObjectIdentifierGeneric::decode): Deleted.
* Source/WebCore/Modules/cache/DOMCacheIdentifier.h:
(): Deleted.
* Source/WebCore/Modules/fetch/FetchIdentifier.h:
(): Deleted.
* Source/WebCore/Modules/filesystemaccess/FileSystemHandleIdentifier.h:
(): Deleted.
* Source/WebCore/Modules/filesystemaccess/FileSystemSyncAccessHandleIdentifier.h:
(): Deleted.
* Source/WebCore/Modules/filesystemaccess/WorkerFileSystemStorageConnectionCallbackIdentifier.h:
(): Deleted.
* Source/WebCore/Modules/mediastream/RTCRtpScriptTransformer.h:
* Source/WebCore/Modules/permissions/MainThreadPermissionObserverIdentifier.h:
(): Deleted.
* Source/WebCore/Modules/push-api/PushSubscriptionIdentifier.h:
(): Deleted.
* Source/WebCore/Modules/speech/SpeechRecognitionConnectionClientIdentifier.h:
(): Deleted.
* Source/WebCore/Modules/web-locks/WebLockIdentifier.h:
(): Deleted.
* Source/WebCore/Modules/websockets/WebSocketIdentifier.h:
(): Deleted.
* Source/WebCore/accessibility/AXCoreObject.h:
* Source/WebCore/dom/BroadcastChannelIdentifier.h:
(): Deleted.
* Source/WebCore/dom/ElementIdentifier.h:
(): Deleted.
* Source/WebCore/dom/PortIdentifier.h:
(): Deleted.
* Source/WebCore/editing/TextManipulationItemIdentifier.h:
(): Deleted.
* Source/WebCore/editing/TextManipulationToken.h:
* Source/WebCore/editing/cocoa/AttributedString.h:
* Source/WebCore/page/GlobalWindowIdentifier.h:
(): Deleted.
* Source/WebCore/page/SecurityOriginData.h:
(): Deleted.
* Source/WebCore/page/cocoa/ImageOverlayDataDetectionResultIdentifier.h:
(): Deleted.
* Source/WebCore/platform/MediaSessionGroupIdentifier.h:
(): Deleted.
* Source/WebCore/platform/MediaUniqueIdentifier.h:
(): Deleted.
* Source/WebCore/platform/ProcessIdentifier.h:
(): Deleted.
* Source/WebCore/platform/graphics/ImageDecoderIdentifier.h:
(): Deleted.
* Source/WebCore/platform/graphics/InbandGenericCueIdentifier.h:
(): Deleted.
* Source/WebCore/platform/graphics/LayerHostingContextIdentifier.h:
(): Deleted.
* Source/WebCore/platform/graphics/RenderingResourceIdentifier.h:
(): Deleted.
* Source/WebCore/platform/graphics/mac/DisplayConfigurationMonitor.h:
* Source/WebCore/platform/mediastream/RTCDataChannelLocalIdentifier.h:
(): Deleted.
* Source/WebCore/platform/mediastream/RealtimeMediaSourceIdentifier.h:
(): Deleted.
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCSocketIdentifier.h:
(): Deleted.
* Source/WebCore/platform/text/TextCheckingRequestIdentifier.h:
(): Deleted.
* Source/WebCore/workers/service/ServiceWorkerIdentifier.h:
(): Deleted.
* Source/WebCore/workers/service/ServiceWorkerTypes.h:
(): Deleted.
* Source/WebCore/workers/service/background-fetch/BackgroundFetchRecordIdentifier.h:
(): Deleted.
* Source/WebCore/workers/shared/SharedWorkerIdentifier.h:
(): Deleted.
* Source/WebCore/worklets/WorkletGlobalScope.h:
* Source/WebKit/GPUProcess/media/RemoteMediaResourceIdentifier.h:
(): Deleted.
* Source/WebKit/GPUProcess/media/RemoteSourceBufferIdentifier.h:
(): Deleted.
* Source/WebKit/GPUProcess/media/TrackPrivateRemoteIdentifier.h:
(): Deleted.
* Source/WebKit/NetworkProcess/CustomProtocols/LegacyCustomProtocolID.h:
(): Deleted.
* Source/WebKit/NetworkProcess/Downloads/DownloadID.h:
(): Deleted.
* Source/WebKit/NetworkProcess/PrivateClickMeasurement/cocoa/PrivateClickMeasurementNetworkLoaderCocoa.mm:
(): Deleted.
* Source/WebKit/Platform/IPC/Connection.h:
* Source/WebKit/Platform/IPC/DaemonCoders.cpp:
(WebKit::Daemon::Coder<WebCore::PushSubscriptionIdentifier>::encode):
(WebKit::Daemon::Coder<WebCore::PushSubscriptionIdentifier>::decode):
* Source/WebKit/Scripts/generate-serializers.py:
(Template):
(Template.__init__):
(Template.forward_declaration):
(Template.specialization):
(SerializedType.__init__):
(SerializedType.should_skip_forward_declare):
(one_argument_coder_declaration):
(argument_coder_declarations):
(generate_header):
(construct_type):
(construct_type.is):
(generate_one_impl):
(generate_impl):
(parse_serialized_types):
* Source/WebKit/Scripts/webkit/tests/GeneratedSerializers.cpp:
(IPC::ArgumentCoder<WebKit::TemplateTest<WebKit::Fabulous>>::encode):
(IPC::ArgumentCoder<WebKit::TemplateTest<WebKit::Fabulous>>::decode):
(IPC::ArgumentCoder<WebKit::TemplateTest<WebCore::Amazing>>::encode):
(IPC::ArgumentCoder<WebKit::TemplateTest<WebCore::Amazing>>::decode):
(IPC::ArgumentCoder<WebKit::TemplateTest<JSC::Incredible>>::encode):
(IPC::ArgumentCoder<WebKit::TemplateTest<JSC::Incredible>>::decode):
(IPC::ArgumentCoder<WebKit::TemplateTest<Testing::StorageSize>>::encode):
(IPC::ArgumentCoder<WebKit::TemplateTest<Testing::StorageSize>>::decode):
* Source/WebKit/Scripts/webkit/tests/GeneratedSerializers.h:
* Source/WebKit/Scripts/webkit/tests/SerializedTypeInfo.cpp:
(WebKit::allSerializedTypes):
* Source/WebKit/Scripts/webkit/tests/TestSerializedType.serialization.in:
* Source/WebKit/Shared/ContentWorldShared.h:
(): Deleted.
* Source/WebKit/Shared/Extensions/WebExtensionControllerIdentifier.h:
(): Deleted.
* Source/WebKit/Shared/ShapeDetectionIdentifier.h:
(): Deleted.
* Source/WebKit/Shared/WTFArgumentCoders.serialization.in:
* Source/WebKit/Shared/WebGPU/WebGPUIdentifier.h:
(): Deleted.
* Source/WebKit/Shared/WebsiteData/QuotaIncreaseRequestIdentifier.h:
(): Deleted.
* Source/WebKit/Shared/XR/XRDeviceIdentifier.h:
(): Deleted.
* Source/WebKit/UIProcess/UserMediaPermissionRequestManagerProxy.h:
* Source/WebKit/UIProcess/WebURLSchemeHandler.cpp:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/WebProcess/GPU/graphics/GraphicsContextGLIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/graphics/RenderingBackendIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/media/RemoteAudioHardwareListenerIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/media/RemoteAudioSessionIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/media/RemoteCDMIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/media/RemoteCDMInstanceIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/media/RemoteCDMInstanceSessionIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/media/RemoteLegacyCDMIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/media/RemoteLegacyCDMSessionIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/media/RemoteMediaSourceIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/media/RemoteRemoteCommandListenerIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/media/RemoteVideoFrameIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/webrtc/AudioMediaStreamTrackRendererInternalUnitIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/webrtc/MediaRecorderIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/webrtc/SampleBufferDisplayLayerIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/webrtc/VideoDecoderIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/GPU/webrtc/VideoEncoderIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCResolverIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/Storage/RetrieveRecordResponseBodyCallbackIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/WebStorage/StorageAreaImplIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/WebStorage/StorageAreaMapIdentifier.h:
(): Deleted.
* Source/WebKit/WebProcess/WebStorage/StorageNamespaceIdentifier.h:
(): Deleted.

Canonical link: https://commits.webkit.org/268623@main
